pkg/apis/core/validation/validation.go

```diff
@@ -4154,6 +4152,48 @@ func validateNodeConfigSource(source *core.NodeConfigSource, fldPath *field.Path
 	return allErrs
 }
 
+// validation specific to Node.Spec.ConfigSource.ConfigMap
+func validateConfigMapNodeConfigSourceSpec(source *core.ConfigMapNodeConfigSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// name, namespace, uid, and kubeletConfigKey must all be non-empty for ConfigMap
+	if source.Name == "" {
```

Comment:
move the name/namespace/kubeletconfigkey requirements into validateConfigMapNodeConfigSource and skip name format validation for each if a required error is added, so you don't end up with two errors per field ("this field is required" AND "this field's format is invalid")

Reply:
done